src/Options/OptionsDetails.hpp

```diff
@@ -131,68 +167,83 @@ struct has_upper_bound_on_size<
     S, cpp17::void_t<decltype(std::declval<S>().upper_bound_on_size())>>
     : std::true_type {};
 
-struct print {
-  explicit print(const int max_label_size) noexcept
-      : max_label_size_(max_label_size) {}
+template <typename OptionList>
+struct print;
 
-  using value_type = std::string;
-  template <typename T, Requires<has_default<T>::value> = nullptr>
+template <typename T, typename OptionList, typename = std::nullptr_t>
+struct print_impl {
+  // T is a group here
+  using subgroups_t = typename options_in_group<OptionList, T>::type;
+  static std::string apply(const int max_label_size) noexcept {
+    std::ostringstream ss;
+    ss << "  " << std::setw(max_label_size + 2) << std::left
+       << (name<T>() + ":") << T::help << "\n\n";
+    ss << tmpl::for_each<subgroups_t>(print<subgroups_t>{max_label_size}).value;
+    ss << "\n";
+    return ss.str();
+  }
+};
+
+template <typename T, typename OptionList>
+struct print_impl<T, OptionList,
+                  Requires<tmpl::list_contains_v<OptionList, T>>> {
+  // T not a group here, but an option
+  template <typename U, Requires<has_default<U>::value> = nullptr>
```

Comment:
Maybe here we can also go against my terrible habits and name `U` something like `LTag` or `LocalTag` or something with `Tag` in the name?